src/flint/test/test.py

```diff
@@ -1584,6 +1584,212 @@ def test_pickling():
+    assert F163_copy == F163
+    assert Fbig != F163
+    assert hash(F163_copy) == hash(F163)
+    assert hash(Fbig) != hash(F163), f"{hash(Fbig)}, {hash(F163)}"
+    assert F163_copy != Fbig
+    assert F163_copy != "A"
+
+    assert str(F163) == "fmpz_mod_ctx(163)"
+    assert repr(F163) == "Context for fmpz_mod with modulus: 163"
+
+    # Type tests
+
+    assert raises(lambda: fmpz_mod(1, "AAA"), ValueError)
+
+
+    # Rich comparisons 
+    assert raises(lambda: F163(123) > 0, TypeError)
+    assert raises(lambda: F163(123) >= 0, TypeError)
+    assert raises(lambda: F163(123) < 0, TypeError)
+    assert raises(lambda: F163(123) <= 0, TypeError)
+
+    assert (F163(123) == 123) is True
+    assert (F163(123) == 123 + 163) is True
+    assert (F163(123) == 1) is False
+    assert (F163(123) != 1) is True
+    assert (F163(123) == F163(123)) is True
+    assert (F163(123) == F163(123 + 163)) is True
+    assert (F163(123) == F163(1)) is False
+    assert (F163(123) != F163(1)) is True
+
+    assert (hash(F163(123)) == hash(123)) is True
+    assert (hash(F163(F163(123))) == hash(123)) is True
+    assert (hash(F163(123)) == hash(1)) is False
+    assert (hash(F163(123)) != hash(1)) is True
+    assert (hash(F163(123)) == hash(F163(123))) is True
+    assert (hash(F163(123)) == hash(F163(123 + 163))) is True
+    assert (hash(F163(123)) == hash(F163(1))) is False
+    assert (hash(F163(123)) != hash(F163(1))) is True
+
+    # Is one, zero, canoncial
+    assert (F163(0) == 0) is True
+    assert F163(0).is_zero() is True
+    assert not F163(0)
+    assert not F163(163)
+    assert F163(1).is_one() is True
+    assert F163(164).is_one() is True
+    assert F163(1).is_one() is True
+    assert F163(2).is_one() is False
+
+    # int, str, repr
+    assert str(F163(11)) == "11"
+    assert str(F163(-1)) == "162"
+    assert repr(F163(11)) == "fmpz_mod(11, 163)"
+    assert repr(F163(-1)) == "fmpz_mod(162, 163)"
+
+    assert F163(5).__pos__() == F163(5)
+
+    # Arithmetic tests
+
+    # Negation
+    assert -F163(123) == F163(-123) == (-123 % 163)
+    assert -F163(1) == F163(-1) == F163(162)
+
+    # Addition
+    assert F163(123) + F163(456) == F163(123 + 456)
+    assert F163(123) + F163_copy(456) == F163(123 + 456)
+    assert F163(123) + F163(456) == F163_copy(123 + 456)
+    assert raises(lambda: F163(123) + "AAA", TypeError)
+
+    assert F163(123) + F163(456) == F163(456) + F163(123)
+    assert F163(123) + 456 == F163(123 + 456)
+    assert 456 + F163(123) == F163(123 + 456)
+    assert F163(123) + fmpz(456) == F163(456) + F163(123)
+    assert raises(lambda: F163(123) + Fbig(456), ValueError)
+
+    test_inplace = F163(123) 
+    test_inplace += F163(456) 
+    assert test_inplace == F163(123 + 456)
+
+    test_inplace = F163(123) 
+    test_inplace += 456
+    assert test_inplace == F163(123 + 456)
+
+    test_inplace = F163(123) 
+    test_inplace += fmpz(456)
+    assert test_inplace == F163(123 + 456)
```

Comment:
The in-place methods are removed so the references to `inplace` here are confusing. We don't need to separately test `+=` from `+` if we are not implementing the `__iadd__` etc methods. If we did want to test anything with `+=` then it would be to assert that the original object was not mutated which is not what this tests.